### ISSUE TITLE
"Only variables should be passed by reference" - fileglobalgc.php

### ISSUE THREAD

**Issue by fukawi2** [owncloud/core#14696]
### Steps to reproduce
1. Upgrade to `owncloud-server-8.0.1-11.1.noarch` on CentOS 7
2. Run cron.php using system cron.

### Expected behaviour
No PHP errors or warnings.

### Actual behaviour
Each invocation of cron.php results in warnings:

    {"reqId":"3833c91003ad6f7edbd63762bb062ccf","remoteAddr":"","app":"PHP","message":"Only variables should be passed by reference at \/var\/www\/html\/owncloud\/lib\/private\/cache\/fileglobalgc.php#50","level":3,"time":"2015-03-05T13:30:01+11:00"}
    {"reqId":"3833c91003ad6f7edbd63762bb062ccf","remoteAddr":"","app":"PHP","message":"chmod(): Operation not permitted at \/var\/www\/html\/owncloud\/lib\/private\/log\/owncloud.php#85","level":3,"time":"2015-03-05T13:30:01+11:00"}

Resolved (temporarily) by changing line 50 from:

    array_walk($this->getExpiredPaths($cacheDir, $now), 'unlink');

To:

    $expiredPaths = $this->getExpiredPaths($cacheDir, $now);
    array_walk($expiredPaths, 'unlink');

### Server configuration
**Operating system**:

CentOS 7

**Web server:**

Apache 2.4 => `httpd-2.4.6-19.el7.centos.x86_64`

**Database:**

MySQL => `mariadb-server-5.5.41-2.el7_0.x86_64`

**PHP version:**

`php-5.4.16-23.el7_0.3.x86_64`

**ownCloud version:** (see ownCloud admin page)

8.0.1 (stable) (RPM: owncloud-server-8.0.1-11.1.noarch)

**Updated from an older ownCloud or fresh install:**

Updated using yum from http://download.opensuse.org/repositories/isv:/ownCloud:/community/CentOS_CentOS-7/

**List of activated apps:**

Activity, Deleted files, External storage support, First Run Wizard, Mail Template Editor, PDF Viewer, Pictures, Server-side Encryption, Share Files, Text Editor, Updater, Versions, Video Viewer.

**The content of config/config.php:**

```
$CONFIG = array (
  'instanceid' => 'oc89b470161f',
  'passwordsalt' => 'passwordsalt',
  'trusted_domains' => 
  array (
    0 => 'owncloud.mydomain.net',
  ),
  'datadirectory' => '/var/www/html/owncloud/data',
  'version' => '8.0.1.1',
  'dbtype' => 'mysql',
  'dbname' => 'ownclouddb',
  'dbhost' => 'localhost',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'ownclouduser',
  'dbpassword' => 'dbpassword',
  'installed' => true,
  'mail_from_address' => 'owncloud',
  'mail_smtpmode' => 'smtp',
  'mail_domain' => 'mydomain.net',
  'logtimezone' => 'Australia/Melbourne',
  'logfile' => '/var/log/owncloud.log',
  'loglevel' => '2',
  'log_authfailip' => true,
  'overwritehost' => 'owncloud.mydomain.net',
  'mail_smtphost' => 'localhost',
  'mail_smtpport' => '25',
  'forcessl' => true,
  'theme' => '',
  'secret' => 'secret',
  'maintenance' => false,
);
```

**Are you using external storage, if yes which one:** local/smb/sftp/...

No.

**Are you using encryption:** yes/no

Yes.

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

No.

**Comment by motorsense** [comment]
I confirm error on ubuntu 14.04

**Comment by nickvergessen** [comment]
Backport has been merged, closing

**Comment by milprog** [comment]
confirm this error on centos-6 w/apache 2.4 (SCL) and php-fpm 5.5 (remi) using oc 8.0.2-14.1 installed via rpm.

**Comment by Xenopathic** [comment]
The fix will be in 8.0.3
